src/clarity_vm/clarity.rs

```diff
@@ -887,6 +894,141 @@ impl<'a, 'b> ClarityBlockConnection<'a, 'b> {
+                                    ("reward-cycle".into(), Value::UInt(transform.cycle as u128)),
+                                    ("index".into(), Value::UInt(transform.index as u128)),
+                                ])
+                                .unwrap();
+                                let mut value = transform.original_entry;
+                                match value.data_map.entry("total-ustx".into()) {
+                                    std::collections::btree_map::Entry::Vacant(_) => {
+                                        panic!("FATAL: existing entry must have total-ustx entry")
+                                    }
+                                    std::collections::btree_map::Entry::Occupied(mut e) => {
+                                        *e.get_mut() = Value::UInt(transform.new_amount)
+                                    }
+                                };
+
+                                // Is it safer to specify the map_type? I think using unknown_descriptor is safer, because
+                                //  that actually loads the stored descriptor and then checks against
+                                //  that.
+                                //
+                                // let map_type = DataMapMetadata {
+                                //     key_type: TupleTypeSignature::try_from(vec![
+                                //         ("reward-cycle".into(), TypeSignature::UIntType),
+                                //         ("index".into(), TypeSignature::UIntType),
+                                //     ]).unwrap().into(),
+                                //     value_type: TupleTypeSignature::try_from(vec![
+                                //         ("pox-addr".into(), TupleTypeSignature::try_from(
+                                //             vec![("version".into(), BUFF_1.clone()),
+                                //                  ("hashbytes".into(), BUFF_32.clone())]
+                                //         ).unwrap().into()),
+                                //         ("total-ustx".into(), TypeSignature::UIntType),
+                                //         ("stacker".into(), TypeSignature::new_option(TypeSignature::PrincipalType).unwrap()),
+                                //     ]).unwrap().into(),
+                                // };
+
+                                exec_env
+                                    .global_context
+                                    .database
+                                    .set_entry_unknown_descriptor(
+                                        &boot_code_id(POX_2_NAME, mainnet),
+                                        "reward-cycle-pox-address-list",
+                                        key.into(),
+                                        value.into(),
+                                    )
+                                    .expect("FATAL: Failed to update chainstate db");
```

Comment:
> Why you're not upfront about what this PR propose and explain in simple words (so that everyone could understand it) that you want to alter the on-chain data in such way that it will be impossible to trace when or how it happened, because there won't be any trace of it in any block or transaction?
In other words it will be like transferring STX from one account to another without transaction - simply hard-coded in node code.

I believe this PR is pretty up-front about what it is doing. But, this PR is probably not the right forum to discuss the hard-fork resolution. That would be the SIP, where, rather than having to talk about what the code is doing, the discussion can be about the spec for the code. This PR is simply trying to implement what is proposed in that SIP. If the SIP process goes a different route, then this PR will either be closed or changed to follow what is proposed there.

Secondly, none of the changes in this PR are impossible to trace -- the changes here are applied using state read directly out of the node's chain state. Anyone can run a node that verifies this state. `pox-2` data is already modified by the stacks-node without corresponding transactions: auto-unlocks alter the contract map state, and interacts with the rest of the node in other special ways (i.e. the `special` module). I do agree, though, that other options would avoid modifying the contract state beyond what is already done. I would suggest again the SIP process.

>  Deploy PoX-3 contract, enable burn cycle if it is necessary and move on.

Ultimately, any hard fork performs unusual modification of blockchain state. Enabling and switching to pox-3 would still require this. There's key questions about the feasibility of different approaches, and I'd encourage open dialogue about the approaches. However, this PR is probably a less visible location than others, and it's easier for more people to weigh in on the proposal.

Reply:
@LNow as kantai mentioned, https://github.com/stacksgov/sips/pull/127  is a better place to discuss the why. we should leave this PR for the work related to that proposal.